modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/tree/SingleBarrierKnockoutFunction.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.tree;
+
+import java.util.Arrays;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.product.fx.BarrierType;
+
+/**
+ * Single barrier knock-out option function.
```

Comment:
Probably worth mentioning that there is no equivalent "knock-in", but the knock-in are price by "in-out" parity. To help user looking for a class in the code.